server/src/main/scala/org/http4s/server/staticcontent/FileService.scala

```diff
@@ -86,13 +85,14 @@ object FileService {
               case _ => OptionT.none
             }
           resolvedPath
-            .semiflatMap(path => F.delay(path.toRealPath(LinkOption.NOFOLLOW_LINKS)))
+            .flatMapF(path =>
+              F.delay(if (Files.exists(path)) Some(path.toRealPath(LinkOption.NOFOLLOW_LINKS))
```

Comment:
Can we add the same LinkOption to the `exists` call to make it consistent with the `toRealPath`?

Reply:
Done - thank you for pointing it out.